Add general solution for the PDEs to the docstrings

nicoguaro opened the pull request:
<!-- Please give this pull request a descriptive title. Pull requests with descriptive titles are more likely to receive reviews. Describe what you changed! A title that only references an issue number is not descriptive. -->

<!-- If this pull request fixes an issue please indicate which issue by typing "Fixes #NNNN" below. -->

gxyd commented on sympy/solvers/pde.py:
```diff
@@ -662,18 +677,20 @@ def pde_1st_linear_constant_coeff(eq, func, order, match, solvefun):
 
-    2] `\eta` as the constant in the solution to the differential equation
-    `\frac{dy}{dx} = -\frac{b}{a}`
+    2. `\eta` as the constant in the solution to the differentia
```
typo? `differentia`

nicoguaro replied:
@gxyd, yes, it seems so. Thanks for the catch.

nicoguaro committed:
Add general solution for the PDEs to the docstrings

nicoguaro committed:
Correct typo in docstring

gxyd commented:
@nicoguaro can you please merge master branch in your PR?

nicoguaro commented:
I think I just did it. Let me know if that works

gxyd commented:
It **doesn't** see to have been merged.

nicoguaro commented:
What should I do then?

gxyd commented:
I can do that for you, that wouldn't be a problem. I just thought may want to do that on your own. Let me know if you want me to fix this for you.

nicoguaro commented:
Well, I am not well versed in the use of Git. I thought that I had everything done when asked for this PR. But it seems that it is not true.

Yes, please proceed.

gxyd commented:
I'll review the PR tomorrow (its a little late already) and get back to you.

gxyd commented on sympy/solvers/pde.py:
```diff
@@ -496,7 +496,12 @@ def pde_1st_linear_constant_coeff_homogeneous(eq, func, order, match, solvefun):
 
-    The general solution is of the form::
+    The general solution is of the form:
+    
```
There is trailing white space here, remove that please.

gxyd commented on sympy/solvers/pde.py:
```diff
@@ -662,18 +677,20 @@ def pde_1st_linear_constant_coeff(eq, func, order, match, solvefun):
-    where `a(x, y)`, `b(x, y)`, `c(x, y)` and `G(x, y)` are arbitrary functions
-    in `x` and `y`. This PDE is converted into an ODE by making the following transformation.
+    where `a(x, y)`, `b(x, y)`, `c(x, y)` and `G(x, y)` are arbitrary
+    functions     in `x` and `y`. This PDE is converted into an ODE by
```
Also remove unnecessary space after 'functions'.

nicoguaro replied:
What do you mean by "unnecessary space after functions"?

gxyd replied:
There is some extra space after the word 'functions' in the beginning of this line, that isn't needed.

nicoguaro replied:
Oh, I see. I'll fix it

nicoguaro replied:
Oh, I see. I'll fix it. Are we suppose to correct the source code to comply with PEP8?

gxyd replied:
I am not sure if this is included in PEP8 or not. But since the current tests on travis fail because of this. That's why I suggested to change this.

nicoguaro replied:
Ok, I think I made those changes.

nicoguaro committed:
Remove trailing space in pde.py

nicoguaro committed:
Remove trailing space in pde.py

nicoguaro commented:
@gxyd is there something else that need to be done with this?

asmeurer committed:
Merge pull request #15879 from martin-ueding/master

Remove excess indentation in docstring

smichr committed:
Merge pull request #15860 from oscargus/statscleanup

Stats: Removed unused imports and improved documentation structure

smichr committed:
Merge pull request #15814 from oscargus/cleanupprinting

Fixed imports, comments, and some bugs in printing

smichr committed:
Merge pull request #15875 from sylee957/fix_add_is_zero

[WIP] Fixing is_zero for complex numbers in Add

Commit:
add test_issue_15872

oscargus committed:
Corrected ordering to historical

gdp1807 committed:
Improved docs of is_convex

Docs of is_convex has been improved
and the reviews have been taken into
account.

gdp1807 committed:
Merge branch 'concave_convex' of https://github.com/czgdp1807/sympy into concave_convex

oscargus committed:
Merge pull request #15882 from ethankward/issue_15872

Add test for matrix rank issue

oscarbenjamin committed:
Change Example to Examples in docstring

oscarbenjamin committed:
Merge pull request #15876 from oscargus/rootnotationordering

Moved root_notation to correct alphabetical order

oscarbenjamin committed:
Merge pull request #15815 from oscargus/cleanupseries

Clean up of series, combinatorics, codegen, and ntheory

oscargus committed:
Fixed LaTeX printing of bold MatrixSymbol

oscargus committed:
Fixed pretty printing of bold MatrixSymbol

oscargus committed:
Corrected three and four dots above accents and put accents in middle of strings

oscargus committed:
Select a sub-script for test that is supported in Python 2.7

oscargus committed:
Commented out problematic 2.7 doctest

oscargus committed:
Removed example as doctest didn't work for 2.7

smichr committed:
Merge pull request #15846 from sharma-kunal/master

Integration of Polygon gives no len() error solved

jmig5776 committed:
Update

sharma-kunal committed:
NotImplementedError if nonlinsolve can't solve

sharma-kunal committed:
Update solveset.py

sharma-kunal committed:
Update solveset instead of list using any()

sharma-kunal committed:
Added test case

oscargus committed:
Improved simplification of Piecewise

jksuom committed:
Merge pull request #15895 from sharma-kunal/master

NotImplementedError if nonlinsolve can't solve equations in real domain

jmig5776 committed:
Update

jmig5776 committed:
Update

shiksha11 committed:
added root_notation flag in mathml.py

shiksha11 committed:
made root_notation true by default.

gdp1807 committed:
improved docs

oscargus committed:
Fixed issue pointed out in review

gdp1807 committed:
Update crypto.py

smichr committed:
Update test_solvers.py

smichr committed:
Update test_solvers.py

gdp1807 committed:
replaced _pow_mod with pow

oscargus committed:
Fixed sloppy copy-paste causing an issue

bavishhh committed:
Add __init__ to MathMLPresentationPrinter Class

bavishhh committed:
Refactor code

jmig5776 committed:
Update

smichr committed:
Merge pull request #15735 from jmig5776/fix_solve

fixes solve for the given case

oscargus committed:
Merge pull request #15906 from Bavish2201/patch_15877_mathml

Add __init__ to MathMLPresentationPrinter

Commit:
Update factor_.py

This improvement makes the precomputation of small_trailing array more than 100 times faster

jksuom committed:
Merge pull request #15908 from adhoc-king/patch-3

Improved precomputation of small_trailing array

Commit:
Updated trailing function

Tweaked the trailing function, performance improvement is not much for random numbers (~1.05 times faster) but it is 2.5 times faster for powers of 2

Commit:
Update factor_.py

Replace tabs with spaces which caused error in previous patch

Commit:
update docs

oscargus committed:
Corrected test

Geektrovert committed:
Improve docstring in sympy/algebras/quaternion.py -  Attemp on: sympy/sympy#15370

Commit:
Initial WhiteSource configuration file

GaurangTandon committed:
made all as Examples; made === consistent; also no trailing ws

GaurangTandon committed:
Merge branch '15370-patch-3' of https://github.com/GaurangTandon/sympy into 15370-patch-3

GaurangTandon committed:
made == consistent; added newlines and spacing

oscargus committed:
Merge pull request #15750 from GaurangTandon/15370-patch-3

[algebras/quaternion.py] Docstrings added

oscargus committed:
Merge pull request #15910 from divyanshu132/fix_stats

Update stats: Add discrete types and Rademacher

oscargus committed:
Merge pull request #15658 from GaurangTandon/15370-patch-2

#15370 patch 2

Geektrovert committed:
Fix typos. Update docstrings.

Geektrovert committed:
Resolve merge conflict

Geektrovert committed:
Fix 'D0001 line too long' warning in doc/src/modules/stats.rst

RituRajSingh878 committed:
Merge branch 'master' into lambdify

shiksha11 committed:
Merge branch 'master' into newbranch2

Commit:
Update factor_.py

Added a comment

oscargus committed:
Added negated property to Relational and BooleanAtom

shiksha11 committed:
added test for root_notation flag.

smichr committed:
keep binary method in trailing for numbers larger than 2**300

oscarbenjamin committed:
Merge pull request #15761 from mikelRM/master

Custom otional label in Pauli matrices

oscarbenjamin committed:
Merge pull request #15776 from czgdp1807/concave_convex

Added is_convex in sympy.calculus.util

oscarbenjamin committed:
Merge pull request #15820 from oscargus/cleanupalgebras

Clean up and minor bug fixes

smichr committed:
Merge pull request #15909 from adhoc-king/patch-4

Updated trailing function

shiksha11 committed:
removed white space

oscargus committed:
Corrected code and added notes

oscargus committed:
Improved speed slightly for reversed and negated

smichr committed:
Merge pull request #15924 from oscargus/negated

Added negated property to Relational and BooleanAtom

smichr committed:
Merge pull request #15898 from oscargus/improvedpiecewisesimplify

Improved simplification of Piecewise

oscargus committed:
Added import of S

oscargus committed:
Added import of S. Actually. No fat fingers this time.

Geektrovert committed:
Merge remote-tracking branch 'upstream/master'

Geektrovert committed:
Remove trailing whitespace.

Geektrovert committed:
Remove trailing whitespaces.

oscargus committed:
Remove real-value check from Min and Max when rewriting to Piecewise

ssmi1975 committed:
added `nbits` keyword parameter to `IntQubit`.

ssmi1975 committed:
renamed nbits -> nqubits to follow the existing code

ssmi1975 committed:
fixed usage of IntQubit(), fiexed a bug in IntQubit constructor

oscargus committed:
Fixed test.

oscargus committed:
Good to change both functions and update the tests...

oscargus committed:
How hard can it be? Fixed printer call.

RituRajSingh878 committed:
code cleaning

RituRajSingh878 committed:
Merge branch 'lambdify' of https://github.com/RituRajSingh878/sympy into lambdify

oscargus committed:
Cleanup and added root_notation to content printer

Primarily to get a new Azure run, which seems to have not started last time.

ssmi1975 committed:
added type check of arguments

ssmi1975 committed:
added document.

also fixed an issue (and a test) found while checking doctest.

smichr committed:
Merge pull request #15892 from oscargus/correctlycombinedaccents

Corrected three and four dots above accents and put accents in middle of strings

ssmi1975 committed:
fixed the sentence of the doc and removed a trailing space.

oscargus committed:
Added configuration option for imaginary_unit in the LaTeX printer

smichr committed:
Merge pull request #15835 from RituRajSingh878/lambdify

correct dot product of numpy matrix

smichr committed:
Merge branch 'master' into boldmatrixsymbol

ssmi1975 committed:
removed the trailing space missed in the previous commit

Geektrovert committed:
Create doc/src/modules/algebras.rst. Add algebras.rst to index.rst

Geektrovert committed:
Merge remote-tracking branch 'upstream/master'

Geektrovert committed:
Attempt on fixing build fail

Geektrovert committed:
Minor fixes

ssmi1975 committed:
put `nqubit` in the parameter list.

Geektrovert committed:
Fix docstring spacings.

oscargus committed:
Renamed method and fixed failing test after merge

oscargus committed:
Merge pull request #15901 from shiksha11/newbranch2

changes to add root_notation flag in mathml.py.

oscargus committed:
Merge branch 'master' into imaginaryunitlatex

oscargus committed:
Added test for content printer and root_notation

oscargus committed:
Added type check and tests

oscargus committed:
Added imaginary_unit optional keyword for pretty printer

Geektrovert committed:
Merge remote-tracking branch

Geektrovert committed:
Update docstrings in algebras/quaternion.py

ssmi1975 committed:
changed the confusing parameter name "old_assumptions" -> "kwargs"

jksuom committed:
Merge pull request #15933 from ssmi1975/9136-intqubit

Add "nqubits" parameter to IntQubit

Commit:
fix_sympy_mechanics_for_autolev_users

Commit:
fixed

Geektrovert committed:
Fix typos

oscargus committed:
Fixed incorrect import and ordered imports

oscargus committed:
Added predefined short-cuts for imaginary unit options

Geektrovert committed:
Fix typos

smichr committed:
Merge pull request #15936 from oscargus/imaginaryunitlatex

Added configuration option for imaginary_unit in the LaTeX and pretty printers

smichr committed:
Merge pull request #15917 from Geektrovert/master

Improve docstring in sympy/algebras/quaternion.py and create doc/src/modules/algebras.rst -  Attemp on: #15370

Commit:
matrices

Commit:
update matrices

Commit:
Update matrices.py

czgdp1807 committed:
Merged remote bg_crypto into local

czgdp1807 committed:
removed trivial error

RituRajSingh878 committed:
solve exp and difference of logs  with irrational coefficients

smichr committed:
Merge pull request #15941 from divyanshu132/fix_links

Avoid links to doc.sympy.org in the docs

oscarbenjamin committed:
Use cdnjs.cloudflare.com for Mathjax in docs

jmig5776 committed:
Fixed checkodesol function

Update

Update

Update

Update

use xreplace in sub_func_doit

jmig5776 committed:
Fixed checkodesol function

Update

Update

Update

Update

use xreplace in sub_func_doit

supreet11agrawal committed:
Stats: Now only valid probabilities can be entered in Bernoulli, Coin and FiniteRV

frv_types earlier allowed non-valid probability distribution in case of Bernoulli,
Coin and FiniteRV methods. After the modification, only probabilites with value
in between 0 to 1 can be entered. Also in FiniteRV, earlier a random variable
with non valid probability density was allowed. New changes prohibit such shortcomings.

Example:

Bernoulli

>>> from sympy.stats import Bernoulli, density
>>> from sympy import S
>>> X = Bernoulli('X', S(5)/4) #probability can't be >1 but 5/4 here
 Earlier this worked

Now it gives

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy/stats/frv_types.py", line 200, in Bernoulli
    return rv(name, BernoulliDistribution, p, succ, fail)
  File "sympy/stats/frv_types.py", line 28, in rv
    density = cls(*args)
  File "sympy/stats/frv_types.py", line 172, in __new__
    raise ValueError("'p' must be: 0 <= p <= 1 . p = %s" % str(p))
ValueError: 'p' must be: 0 <= p <= 1 . p = 5/4

Similarily, Coin and FiniteRV modified
Can be tested using invalid probability distribution

For FiniteRV, density's each element should be between 0 and 1 and also
total density should be 1. This is now implemented.

supreet11agrawal committed:
Merge branch 'master' into 15903_probability

RituRajSingh878 committed:
added test for issue #6913

RituRajSingh878 committed:
Merge branch 'master' into test_matrices

jmig5776 committed:
Fixed a bug in eval tan

RituRajSingh878 committed:
cleaning code

supreet11agrawal committed:
Update frv_types.py

jmig5776 committed:
Update

Commit:
fix-Matrices

Commit:
Merge branch 'fix_matrices' of https://github.com/divyanshu132/sympy into fix_matrices

jksuom committed:
Merge pull request #15961 from jmig5776/fix_tan

Fixed simplification of an evaluated tan(3*pi/8)

supreet11agrawal committed:
Update frv_types.py

smichr committed:
Merge pull request #15950 from divyanshu132/fix_matrices

Improve docstring of matrices/matrices.py

supreet11agrawal committed:
Update frv_types.py

RituRajSingh878 committed:
code cleaning

smichr committed:
Merge pull request #15954 from RituRajSingh878/exp1

solve exp and difference of logs  with irrational coefficients

supreet11agrawal committed:
Added tests in frv_types

Added tests to check if entered probability is valid or not.
Methods changed:- Bernoulli and FiniteRV
As coin uses Bernoulli Distribution, hence tests for coin are added as well

FAndersson committed:
polygon: correct typo

Commit:
Merge pull request #15967 from FAndersson/polygon_distance_typo

polygon: correct typo

jksuom committed:
Merge pull request #15958 from supreet11agrawal/15903_probability

Stats: Now only valid probabilities can be entered in Bernoulli, Coin and FiniteRV

oscargus committed:
Fixed merge conflicts

oscargus committed:
Merge branch 'master' into issue_6013

oscargus committed:
Removed non-required aliases

oscargus committed:
Added unit test

oscargus committed:
Updated tests

asmeurer committed:
Merge pull request #15970 from oscargus/asmeurer-spacing

Better LaTeX spacing

smichr committed:
Merge pull request #15932 from oscargus/minmax

Remove real-value check from Min and Max when rewriting to Piecewise

RituRajSingh878 committed:
added test case

asmeurer committed:
Merge pull request #15955 from oscarbenjamin/mathjax_cdn

Use cdnjs.cloudflare.com for Mathjax in docs

oscargus committed:
Merged with master

oscargus committed:
Bold matrix symbol option for MathML (and VectorSymbol printing)

RituRajSingh878 committed:
code cleaning

smichr committed:
Merge pull request #15874 from sylee957/cleanup_matrix

Some misc cleanups for matrix

smichr committed:
Merge pull request #15973 from RituRajSingh878/test_matrices1

added test case #2749

smichr committed:
Merge pull request #15971 from oscargus/bhavya17037-issue_6013

Pretty printing of lerchphi

smichr committed:
Merge pull request #15960 from RituRajSingh878/test_matrices

added test for issue #6913

oscargus committed:
Corrected MathML-presentation symbols for sub and sup

oscargus committed:
Fixed LaTeX-printing of ConditionSet with FiniteSet

ShubhamKJha committed:
Fix for finite fourier series

oscargus committed:
Added correct LaTeX printing for sequences with symbolic limits

smichr committed:
pass order to sub_func_doit by default

This is under the user's control only when `sub_func_doit` is called manually.

smichr committed:
pass order to sub_func_doit by default

This is under the user's control only when `sub_func_doit` is called manually.

smichr committed:
Update ode.py

smichr committed:
Update ode.py

smichr committed:
Update ode.py

smichr committed:
Update ode.py

smichr committed:
Update test_ode.py

smichr committed:
Update test_ode.py

smichr committed:
whitespace fix

smichr committed:
whitespace fix

jmig5776 committed:
Merge branch 'fix_checkodesol' of https://github.com/jmig5776/sympy into fix_checkodesol

asmeurer committed:
Update sympy/series/fourier.py

Co-Authored-By: ShubhamKJha <skjha832@gmail.com>

jmig5776 committed:
Update

asmeurer committed:
Merge pull request #15867 from asmeurer/37travis

Add Python 3.7 to Travis

ShubhamKJha committed:
finite fourier fix check for integral coefficient in sin or cos

smichr committed:
mark checks that still fail and comment out

Commit:
fix-summation

smichr committed:
mark 4th one failing

oscarbenjamin committed:
Comment out failing calls to checksysodesol

shubhamabhang77 committed:
Added mat_delim flag in the mathml printer

alpeshjamgade committed:
updated numbers.py

alpeshjamgade committed:
Update numbers.py

smichr committed:
Merge pull request #15993 from alpeshjamgade/sympy_experimental

updated numbers.py

oscarbenjamin committed:
Add checkodesol to all tests in test_ode.py

oscarbenjamin committed:
Remove already commented failing test line

smichr committed:
Merge pull request #15957 from jmig5776/fix_checkodesol

Fixed checkodesol function

oscargus committed:
Merge pull request #15891 from oscargus/boldmatrixsymbol

Option to make MatrixSymbol print in bold for LaTeX, pretty, and MathML printers

oscargus committed:
Merge branch 'master' into mathmloptions

nimish committed:
Alternative derivative printing

oscargus committed:
Intermediate, almost correct, version

oscargus committed:
Improved presentation printer

oscargus committed:
Improved content printer

oscargus committed:
Adapted and added tests

jksuom committed:
Merge pull request #15990 from divyanshu132/fix_summation

Fixed summation

shubhamabhang77 committed:
some minors changes made

shubhamabhang77 committed:
one extra tab was added by mistake , now removed it

shubhamabhang77 committed:
some minor mistake happened in expression variable , now changed it

ShubhamKJha committed:
fix for more general finiteness checking

oscargus committed:
Merge pull request #15991 from shubhamabhang77/configuration_of_mathml

Added mat_delim flag into the mathml printer

oscarbenjamin committed:
Fix nth_algebric to respect constants from the ODE

Commit:
discrete logarithm

ShubhamKJha committed:
update code

Commit:
discrete logarithm

ShubhamKJha committed:
code update

ShubhamKJha committed:
Merge branch 'fourier_fix' of https://github.com/ShubhamKJha/sympy into fourier_fix

ShubhamKJha committed:
update code

oscargus committed:
Merge pull request #16003 from oscargus/printderivative

Improved printing of multiple derivatives in MathML

oscargus committed:
Merge pull request #15980 from oscargus/latexsequence

Added correct LaTeX printing for sequences with symbolic limits

oscargus committed:
Merge pull request #15978 from oscargus/latexconditionset

Fixed LaTeX-printing of ConditionSet with FiniteSet

oscargus committed:
Merge pull request #15976 from oscargus/mathmloptions

Corrected MathML-presentation symbols for sub and sup

Commit:
discrete logarithm

Commit:
test-discrete-logarithm

akritas committed:
multivariate_resultants improved.

jksuom committed:
Simplify max_degrees and update a test.

Commit:
update license

jksuom committed:
Merge pull request #16009 from divyanshu132/fix-discrete-logarithm

Fix for wrong result from discrete logarithm

asmeurer committed:
Merge pull request #16016 from divyanshu132/fix-license

Update year in license

jksuom committed:
Add monomial_set to examples

jksuom committed:
Merge pull request #16013 from jksuom/multivar-resultants

Improvements of multivariate resultants

ShubhamKJha committed:
Added a dummy class to be expanded upon

czgdp1807 committed:
Simplified code

ShubhamKJha committed:
update code

ShubhamKJha committed:
update code

ShubhamKJha committed:
fixes failing tests

ShubhamKJha committed:
update code

ShubhamKJha committed:
Remove an unreachable part of code

jksuom committed:
Merge pull request #15979 from ShubhamKJha/fourier_fix

Fix for finite fourier series

jksuom committed:
Merge pull request #16022 from ShubhamKJha/code_remove

Remove an unreachable part of code

oscarbenjamin committed:
Merge pull request #15825 from czgdp1807/bg_crypto

[WIP] Added Blum Goldwasser cryptosystem

nicoguaro committed:
Remove trailing spaces in pde.py

Remove trailing spaces at line 500

oscarbenjamin committed:
Merge pull request #16008 from oscarbenjamin/nth_alebraic_constants

Fix nth_algebraic to respect constants from the ODE

nicoguaro committed:
Merge branch 'pde_docstrings' of github.com:nicoguaro/sympy into pde_docstrings

nicoguaro committed:
Fix indentation in pde_linear_variable_coeff

nicoguaro committed:
Fix indentation in pde_linear_variable_coeff

nicoguaro committed:
Merge branch 'pde_docstrings' of github.com:nicoguaro/sympy into pde_docstrings
